server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -118,14 +86,35 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
             return;
         }
 
-        String youtubeResBody = getYoutubeRes(API_KEY, accessToken.toString());
-        int numVideos = Integer.parseInt(req.getParameter("num_videos"));
+        String youtubeResBody; 
+        JsonObject likedVideoRes;
+        JsonArray videos;
+        YoutubeGenres genreAnalysis = new YoutubeGenres();
+
+        // next Page Token should be an empty string for first call
+        JsonPrimitive nextPageToken = new JsonPrimitive("");
+        int round = 0;
+        while (nextPageToken != null) {
+            youtubeResBody = getYoutubeRes(API_KEY, 
+                                            accessToken.toString(), 
+                                            nextPageToken.getAsString());            
+            likedVideoRes = JsonParser.parseString(youtubeResBody).getAsJsonObject();
+
+            if (round  == 0) {
```

Comment:
check instead if `nextPageToken == ""`. This will keep you from needing to keep track of a `round` variable.